gradle/libs.versions.toml

```diff
@@ -1,20 +1,24 @@
 [versions]
 # javalite, protoc and protobufjavautil versions should be in sync while updating and
-# it needs to match the protobuf version which grpc has transitive dependency on.
+# it needs to match the protobuf version which grpc has transitive dependency on, which
+# needs to match the version of grpc that grpc-kotlin has a transitive dependency on.
 android-lint = "31.3.2"
 autovalue = "1.10.1"
-coroutines = "1.6.4"
+coroutines = "1.7.3"
```

Comment:
We need to look at the test failures and make sure none of them come from this change. I doubt it, but just to sanity check.

Reply:
All of the GitHub Actions failures looked to me to just be flakes, unrelated to the coroutines version. There is one persistent GitHub Action that fails, but it's just the "CLA" one, and it's complaining because the PR is so big. So we should be good.